### PR TITLE
Fix for register

### PR DESCRIPTION
**Pull request by ppiotr191**

### CODE REVIEW

File: src/main/java/com/intive/patronage/toz/users/UserActivationController.java [hunk 1]
```diff
@@ -30,29 +28,37 @@
...
     })
     @PreAuthorize("hasAuthority('TOZ')")
-    public Proposal sendRegistrationEmail(@PathVariable UUID id) throws IOException, MessagingException {
-        Proposal proposal = proposalService.findOne(id);
-        this.userActivationService.sendActivationEmail(proposal);
-        return proposal;
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public UserActivationResponseView sendRegistrationEmail(@Valid @RequestBody UserActivationRequestView userActivationRequestView) throws IOException, MessagingException {
+
+        this.userActivationService.sendActivationEmailIfProposalExists(userActivationRequestView.getUuid());
```
Comment (Jakub-Izbicki) [inline src/main/java/com/intive/patronage/toz/users/UserActivationController.java]: I think this should return error if `uuid` is not found. After all, it is admin who is sending these activation emails, and I guess admin should know if the email was actually sent. Unlike when a normal user tries to reset password. Also, above you define swagger `@ApiResponse` errors that can actually never occur in this code (`404`).

File: src/main/java/com/intive/patronage/toz/users/UserActivationController.java [hunk 2]
```diff
@@ -30,29 +28,37 @@
...
             this.proposalService = proposalService;
+            this.messageSource = messageSource;
     }
 
-    @ApiOperation(value = "Send registration email", notes = API_METHOD_NOTES)
-    @GetMapping(value = "/{id}")
+    @ApiOperation(value = "Send registration email with token activation", notes = API_METHOD_NOTES)
     @ApiResponses(value = {
             @ApiResponse(code = 400, message = "Bad request", response = ValidationErrorResponse.class),
             @ApiResponse(code = 404, message = "Proposal not found", response = ErrorResponse.class),
```
Comment (Jakub-Izbicki) [inline src/main/java/com/intive/patronage/toz/users/UserActivationController.java]: This (`404`) can never happen in below implementation.

File: src/main/java/com/intive/patronage/toz/users/UserActivationController.java [hunk 3]
```diff
@@ -30,29 +28,37 @@
...
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public UserActivationResponseView sendRegistrationEmail(@Valid @RequestBody UserActivationRequestView userActivationRequestView) throws IOException, MessagingException {
+
+        this.userActivationService.sendActivationEmailIfProposalExists(userActivationRequestView.getUuid());
+
+        final String message = messageSource.getMessage(
+                ACTIVATE_SEND_TOKEN, null, LocaleContextHolder.getLocale());
+
+        return new UserActivationResponseView(message);
     }
```
Comment (Jakub-Izbicki) [inline src/main/java/com/intive/patronage/toz/users/UserActivationController.java]: Unnecessary space.

File: src/test/java/com/intive/patronage/toz/users/UserActivationControllerTest.java [hunk 4]
```diff
@@ -1,89 +0,0 @@
...
-
-@RunWith(SpringJUnit4ClassRunner.class)
-@SpringBootTest
-@TestPropertySource(
-        properties = {
-                ApiProperties.JWT_SECRET_BASE64,
-                ApiProperties.SUPER_ADMIN_PASSWORD
-        }
-)
-public class UserActivationControllerTest {
```
Comment (mlotysz) [inline src/test/java/com/intive/patronage/toz/users/UserActivationControllerTest.java]: Why do you deleted this test?

**Pull request merged**